core/modules/replica/Controller.h

```diff
@@ -0,0 +1,687 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2017 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_CONTROLLER_H
+#define LSST_QSERV_REPLICA_CONTROLLER_H
+
+/// Controller.h declares:
+///
+/// class ControllerIdentity
+/// class Controller
+///
+/// (see individual class documentation for more information)
+
+// System headers
+#include <map>
+#include <memory>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+// Third party headers
+#include <boost/asio.hpp>
+
+// Qserv headers
+#include "replica/Request.h"
+#include "replica/RequestTypesFwd.h"
+#include "replica/ServiceProvider.h"
+
+// Forward declarations
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+// Forward declarations
+
+class ControllerImpl;
+class Messenger;
+
+/**
+ * The base class for implementing requests registry as a polymorphic
+ * collection to store active requests. Pure virtual methods of
+ * the class will be overriden by request-type-specific implementations
+ * (see struct RequestWrappeImplr<REQUEST_TYPE> in the .cc file) capturing
+ * type-dependant pointer and a callback function.
+ */
+struct ControllerRequestWrapper {
+
+    /// The pointer type for instances of the class
+    typedef std::shared_ptr<ControllerRequestWrapper> pointer;
+
+    /// Destructor
+    virtual ~ControllerRequestWrapper() = default;
+
+    /// This method will be called upon a completion of a request
+    /// to notify a subscriber on the event.
+    virtual void notify ()=0;
```

Comment:
extra space after notify

Reply:
G! I thought I eliminated them all :-) Thank you for the finding!